src/Entity/Cart.php

```diff
@@ -97,4 +103,25 @@ public function setTotal(int $total): void
     {
         $this->total = $total;
     }
+
+    public function getState(): ?State
+    {
+        return $this->state;
+    }
+
+    public function setState(?State $state): self
+    {
+        $this->state = $state;
+
+        return $this;
+    }
+
+    public function getNextState(): string
+    {
+        if ($this->state->getLabel() === State::STATE_IN_PREPARATION) {
+            return State::STATE_READY;
```

Comment:
you could have checked that the current state exists, or if I am sneaky I could set something else and then directly go into taken